spec/models/doorkeeper/access_token_spec.rb

```diff
@@ -143,6 +143,12 @@ module NoGenerate
         subject.resource_owner_id = nil
         expect(subject).to be_valid
       end
+
+      it 'is valid without resource_owner_id' do
+        # For resource owner credentials flow
+        subject.application_id = nil
+        expect(subject).to be_valid
+      end
```

Comment:
So it seems I found the issue. There's a default initializer that rails creates: https://github.com/rails/rails/commit/cdbf68599416e6b248d1d0e5bcbbbf73b90f1f42#diff-1846d4882fa63eb5b1271541995cbfdcR41 which controls this flag.